app/src/main/java/com/github/geohunt/app/model/database/firebase/FirebaseChallenge.kt

```diff
@@ -52,6 +52,19 @@ class FirebaseChallengeRef(
 
 }
 
+/**
+ * From the [DatabaseReference] representing the root of challenge, retrieve the reference for the challenge
+ * with a specific @p cid
+ *
+ * @param cid the id of the challenge to be retrieved
+ * @return [DatabaseReference] corresponding reference
+ */
+internal fun DatabaseReference.getChallengeRefFromId(cid: String) : DatabaseReference
+{
+    return child(cid.substring(0, Location.COARSE_HASH_SIZE))
```

Comment:
I do not think that this is a good idea to add this kind of function. The two are totally unrelated.

Reply:
I believe we should have a single function that retrieve the challengeRef based on the database reference and a challengeId because keeping everything groupiert is good. However, your comment is also right, I will convert it to a standard function instead of doing an extension method. How does it sounds ?

Reply:
Nevermind better idea, I will make it an extensions method of FirebaseDatabase instead